pkg/controller/operators/olm/operator.go

```diff
@@ -158,6 +182,54 @@ func (a *Operator) syncDeployment(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) syncOperatorGroups(obj interface{}) error {
+	op, ok := obj.(*v1alpha2.OperatorGroup)
+	if !ok {
+		log.Debugf("wrong type: %#v\n", obj)
+		return fmt.Errorf("casting OperatorGroup failed")
+	}
+
+	// write namespace matches to status field
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+	op.Status.Namespaces = namespaceList.Items
+
```

Comment:
Once we get here, we want to check and see if the list of namespaces has changed since the last time we ran the selector. If they're the same, exit early here and don't update all of the deployments (no need to, as their annotations wouldn't have changed).